launch/launch/descriptions/executable.py

```diff
@@ -0,0 +1,141 @@
+"""Module for a description of an Executable."""
+
+import os
+import shlex
+import threading
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from launch.some_substitutions_type import SomeSubstitutionsType
+from launch.substitution import Substitution
+from launch.launch_context import LaunchContext
+from launch.utilities import normalize_to_list_of_substitutions
+from launch.utilities import perform_substitutions
+
+_global_process_counter_lock = threading.Lock()
+_global_process_counter = 0  # in Python3, this number is unbounded (no rollover)
```

Comment:
this is going to be a problem until `ExecuteProcess` doesn't internally use this new description, as there is going to be two global counters (and thus, names may collide).

This does not need to be fixed now.

Reply:
I've changed the name to a local-specific name regardless. There's no need for those to conflict.